src/main/webapp/assets/js/feedScript.js

```diff
@@ -605,16 +614,204 @@ function closeModal() {
+  }
+}
+
+/**
+ * Resets all the input backgrounds to modal grey and gets rid of previous error text.
+ * @param {array} formElements
+ * @return {void}
+ */
+function resetMarks(formElements) {
+  for (let i = 0; i < formElements.length - 1; i++) {
+    const elt = formElements[i];
+    elt.style.background = '#1b18181a';
+  }
+  const modalError = document.getElementById('modal-input-error');
+  if (modalError) {
+    modalForm.removeChild(modalError);
+  }
+}
+
+/**
+ * Goes through the invalid inputs and colors them red.
+ * Adds error messages.
+ * @param {array} invalidIds
+ * @param {array} errorMessages
+ * @param {array} formElements
+ * @return {void}
+ */
+function markInvalidInputs(invalidIds, errorMessages, formElements) {
+  resetMarks(formElements);
+  invalidIds.forEach((id) => {
+    const elt = formElements.namedItem(id);
+    if (elt) {
+      elt.style.background = '#ff999966';
     }
+  });
+  if (errorMessages.length > 0) {
+    const modalError = document.createElement('div');
+    modalError.setAttribute('id', 'modal-input-error');
+    modalError.setAttribute('tabindex', '0');
+    modalError.innerHTML += '<ul>';
+    errorMessages.forEach((message) => {
+      modalError.innerHTML += '<li>' + message + '</li>';
+    });
+    modalError.innerHTML += '</ul>';
+    modalForm.insertBefore(modalError, submitModalButton);
   }
+}
+
+/**
+ * Checks if the month and day are a valid date.
+ * @param {array} invalidIds
+ * @param {array} errorMessages
+ * @param {array} formElements
+ * @return {void}
+ */
+function validateModalDate(invalidIds, errorMessages, formElements) {
+  const month = formElements.namedItem('modal-month').value;
+  const day = formElements.namedItem('modal-day').value;
+  const monthDayLengths = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+  if (month > 12 || month < 1) {
+    invalidIds.push('modal-month');
+    errorMessages.push('month must be between 1 and 12');
+  }
+  if (day < 1 || day > monthDayLengths[month - 1] || isBlank(day)) {
+    invalidIds.push('modal-day');
+    errorMessages.push('invalid date');
+  }
+}
+
+/**
+ * Checks if the start and end time are valid.
+ * @param {array} invalidIds
+ * @param {array} errorMessages
+ * @param {array} formElements
+ * @return {void}
+ */
+function validateModalTime(invalidIds, errorMessages, formElements) {
+  const startHour = parseInt(formElements.namedItem('modal-start-hour').value, 10);
+  const startMinute = parseInt(formElements.namedItem('modal-start-minute').value, 10);
+  const startAMorPM = parseInt(formElements.namedItem('start-am-or-pm').value, 10);
+  const endHour = parseInt(formElements.namedItem('modal-end-hour').value, 10);
+  const endMinute = parseInt(formElements.namedItem('modal-end-minute').value, 10);
+  const endAMorPM = parseInt(formElements.namedItem('end-am-or-pm').value, 10);
+
+  // Check if the hours fall between 1-12.
+  if (startHour < 1 || startHour > 12 || isNaN(startHour)) {
```

Comment:
I would put isNan first. In some languages, if you try comparing a non-number with a number, you might error out so you want to check that it's a number first.

Reply:
Apply this below as well.

Reply:
Great point!